### PR TITLE
Painless: Restructure Definition/Whitelist

### PR DESCRIPTION
**Pull request by jdconrad**
Changes look long, but it's just a combination of renaming several different related classes.

The following changes are made:

* Create lookup package
* rename Definition to PainlessLookup and move to lookup package
* rename Definition.Method to PainlessMethod
* rename Definition.MethodKey to PainlessMethod
* rename Definition.Field to PainlessField
* rename Definition.Struct to PainlessClass
* rename Definition.Cast to PainlessCast
* rename Whitelist.Struct to WhitelistClass
* rename Whitelist.Constructor to WhitelistConstructor
* rename Whitelist.Method to WhitelistMethod
* rename Whitelist.Field to WhitelistField

### CODE REVIEW

File: modules/lang-painless/src/main/java/org/elasticsearch/painless/Def.java [hunk 1]
```diff
@@ -322,30 +324,30 @@ static MethodHandle lookupMethod(Definition definition, Lookup lookup, MethodTyp
       * This is just like LambdaMetaFactory, only with a dynamic type. The interface type is known,
       * so we simply need to lookup the matching implementation method based on receiver type.
       */
-    static MethodHandle lookupReference(Definition definition, Lookup lookup, String interfaceClass,
-            Class<?> receiverClass, String name) throws Throwable {
-         Class<?> interfaceType = definition.getJavaClassFromPainlessType(interfaceClass);
-         Method interfaceMethod = definition.getPainlessStructFromJavaClass(interfaceType).functionalMethod;
+    static MethodHandle lookupReference(PainlessLookup painlessLookup, Lookup lookup, String interfaceClass,
```
Comment (rjernst) [inline modules/lang-painless/src/main/java/org/elasticsearch/painless/Def.java]: I know this was already existing, but can you rename the MethodHandle.Lookup parameter to `methodLookup`? (maybe also don't static import Lookup)

**Review by rjernst — APPROVED** [review]
LGTM

File: modules/lang-painless/src/main/java/org/elasticsearch/painless/Def.java (hunk 1 above)
Comment (jdconrad) [reply modules/lang-painless/src/main/java/org/elasticsearch/painless/Def.java]: Done.

**Comment by jdconrad** [comment]
@rjernst Thanks for the review.  Will commit as soon as CI completes.

**Pull request merged**